kakao/src/main/kotlin/com/agoda/kakao/configurator/ConfigModel.kt

```diff
@@ -0,0 +1,15 @@
+package com.agoda.kakao.configurator
+
+import android.support.test.espresso.DataInteraction
+import android.support.test.espresso.ViewInteraction
+import android.support.test.espresso.web.sugar.Web
+import com.agoda.kakao.delegates.DataInteractionDelegate
+import com.agoda.kakao.delegates.ViewInteractionDelegate
+import com.agoda.kakao.delegates.WebInteractionDelegate
+
+data class ConfigModel(
+    val parentConfig: ConfigModel?,
+    val viewInteractionDelegateFactory: ((ViewInteraction) -> ViewInteractionDelegate),
```

Comment:
Can you please justify the need to use factory pattern here?

Reply:
Yep, maybe I need to create a better name.

Reply:
Renamed to "source" instead of "factory"